profile/src/main/java/rdx/works/profile/domain/GenerateProfileUseCase.kt

```diff
@@ -26,41 +30,41 @@ class GenerateProfileUseCase @Inject constructor(
     @DefaultDispatcher private val defaultDispatcher: CoroutineDispatcher
 ) {
 
-    suspend operator fun invoke(): Profile {
-        return when (val state = profileRepository.profileState.first()) {
-            is ProfileState.Restored -> state.profile
-            else -> withContext(defaultDispatcher) {
-                val profile = Profile.init(
-                    id = UUIDGenerator.uuid().toString(),
-                    deviceInfo = deviceInfoRepository.getDeviceInfo(),
-                    creationDate = InstantGenerator()
-                )
-
-                profileRepository.saveProfile(profile)
-
-                profile
-            }
+    suspend operator fun invoke(mnemonicWithPassphrase: MnemonicWithPassphrase): Profile {
+        val device = deviceInfoRepository.getDeviceInfo()
+        return Profile.init(
+            deviceFactorSource = FactorSource.Device.babylon(
+                mnemonicWithPassphrase = mnemonicWithPassphrase,
+                model = device.model,
+                name = device.name,
+                createdAt = TimestampGenerator(),
+                isMain = true
+            ),
+            creatingDeviceName = device.displayName
+        ).also {
+            profileRepository.saveProfile(it)
         }
     }
 
-    suspend fun initWithBdfsAndAccounts(
-        bdfs: DeviceFactorSource,
+    suspend fun derived(
```

Comment:
i think it's better to separate this in a new usecase

Reply:
It is but I am not planning to do any other refactoring in this PR